### PR TITLE
Add May Preview And May GA changes for Analyze ConversationsAurghob/convmaypreview ga

### LINKED ISSUE #28093: [Azure Cognitive service for language ] API Review
New API Review meeting has been requested.

**Service Name**: Azure Cognitive service for language 
**Review Created By**: Aurgho Bhattacharjee
**Review Date**: 04/09/2024 01:00 PM PT
**Release Plan**: [](https://web.powerapps.com/apps/821ab569-ae60-420d-8264-d7b5d5ef734c?release-plan-id=)
**PR**: https://github.com/quentinRobinson/azure-rest-api-specs/pull/1
**Hero Scenarios Link**: Not Provided
**Core Concepts Doc Link**: Not Provided

**Description**: May preview and GA analyze-conversations API review

Detailed meeting information and documents provided can be accessed [here](https://web.powerapps.com/apps/e41c69de-ef96-44ba-b287-0e012e7c13ae?view=edit&event=b8a7986e-17db-ee11-904d-000d3a3411c3)
For more information that will help prepare you for this review, the requirements, and office hours, visit the documentation [here](https://aka.ms/azsdk/onboarding/restapischedule)

### PR DESCRIPTION
**Pull request by aurghob**
# Data Plane API - Pull Request

Adding May Preview and May GA API for Analyze Conversations API for Cognitive Services Language.
This change is based off an API review done on 4/9 : https://github.com/Azure/azure-rest-api-specs/issues/28093 and takes changes from https://github.com/quentinRobinson/azure-rest-api-specs/pull/1 and https://github.com/aurghob/azure-rest-api-specs/pull/1.

Brief overview of changes:
1. All November Preview API changes carry over to May Preview API (2023-05-15-preview) except for conversational sentiment which is removed.
2. Conv PII is GA with the GA API (2024-05-01).
3. Conv Summarization task stays GA.
4. Note : Custom Conversation summarization stays in preview and not GA.
5. follow-up tasks aspect used in Conv Summarization is not being GA and stays in upcoming preview. It was present in the last Nov Preview API (2023-11-15-preview).
6. GA of summaryLength parameter in abstractive summarization task parameter. (not covered in API review)
7. GA of Recap Summary Aspect (not covered in API review)
8. Removal of Conversational Sentiment as at is no longer available from new GA and preview API and being deprecated (mentioned but not explicitly removed in API review).

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [x] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://github.com/Azure/azure-api-style-guide/blob/main/README.md)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/azapi/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/azapi/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Schedule a data plane REST API spec review](https://aka.ms/azsdk/onboarding/restapischedule)

</details>

<details>
  <summary>Checks stuck in `queued` state?</summary>
If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.  
</details>

### CODE REVIEW

**Review by bidisha-c — APPROVED** [review]

File: specification/cognitiveservices/Language.Conversations/main.tsp [hunk 1]
```diff
@@ -1161,9 +1107,19 @@ model ConversationSummarizationTaskParameters is PreBuiltTaskParameters {
...
+@added(Versions.v2024_05_15_preview)
 model CustomConversationSummarizationTaskParameters is CustomTaskParameters {
-  ...ConversationSummarizationTaskParametersBase;
+  /** It controls the approximate number of sentences in the output summaries. */
+  sentenceCount?: int32;
+
+  /** String index type */
+  stringIndexType?: StringIndexType = StringIndexType.TextElements_v8;
+
+  /** (NOTE: Recommended to use summaryLength over sentenceCount) Controls the approximate length of the output summaries. */
```
Comment (heaths) [inline specification/cognitiveservices/Language.Conversations/main.tsp:1118]: Is this note meant for customers? If so, for docs, I'd consider reformatting this. It just reads like it was meant for internal devs.

**Review by heaths — APPROVED** [review]
Just one nit about docs, but otherwise LGTM.

File: specification/cognitiveservices/Language.Conversations/main.tsp (hunk 1 above)
Comment (aurghob) [reply specification/cognitiveservices/Language.Conversations/main.tsp]: Combined into documentation by adding later in the same sentence.

**Review by mikekistler — APPROVED** [review]
GA API looks good. 👍 

Some concerns with preview API we may revisit in the next review.

**Pull request merged**